src/components/swap/SwapBox.vue

```diff
@@ -130,7 +135,30 @@ export default defineComponent({
     },
     async swapFrom(model) {
       if (model) {
-        // TODO: Call the method that gets the gas fee and calculate swapTo.value
+        this.hasAllowance = false;
+        let balance;
+        // native tokens needs to approval?
+        if (model.type === "NATIVE") {
+          balance = await this.web3.eth.getBalance(this.account);
+
+          // this.swapFrom.balance = this.web3.utils.fromWei(this.account); // todo: why this doesnt work?
+          this.swapFrom.balance = new BigNumber(balance).shiftedBy(
+            -model.decimals || -18
+          );
+        } else {
+          const tokenInst = new this.web3.eth.Contract(
+            ERC20_ABI,
+            model.address
+          );
+          balance = await tokenInst.methods.balanceOf(this.account).call();
+          this.swapFrom.balance = this.web3.utils.fromWei(balance);
+
+          const spenderAddress = "0x7c77704007C9996Ee591C516f7319828BA49d91E"; //
```

Comment:
allowance doesn't apply anymore, i removed it, please check that you merged correctly.

Reply:
removed, sorry, bad refactor